pkg/discovery/service.go

```diff
@@ -158,6 +159,24 @@ func (discovery *Discovery) refresh(ctx context.Context) error {
 		if err != nil {
 			discovery.log.Error("could not put node into cache", zap.String("ID", ping.Id.String()), zap.Error(err))
 		}
+
+		// update email and wallet with correct info
+		_, info, err := discovery.kad.FetchInfo(ctx, &pb.NodeAddress{
+			Address: ping.GetAddress().String(),
+		})
```

Comment:
No need to create new `pb.NodeAddress`. You can use `ping.GetAddress()` directly, i.e.
```go
_, info, err := discovery.kad.FetchInfo(ctx, ping.GetAddress())
```

Reply:
Fixed 👍